mypy/semanal.py

```diff
@@ -317,6 +317,14 @@
 CORE_BUILTIN_CLASSES: Final = ["object", "bool", "function"]
 
 
+# Python has several different scope/namespace kinds with subtly different semantics.
+SCOPE_GLOBAL: Final = 0  # Module top level
+SCOPE_CLASS: Final = 1  # Class body
+SCOPE_FUNC: Final = 2  # Function or lambda
+SCOPE_COMPREHENSION: Final = 3  # Comprehension or generator expression
+SCOPE_TYPE_PARAM: Final = 4  # Python 3.12 new-style type parameter scope (PEP 695)
```

Comment:
The same scoping is used for type alias values and in the future will be used for annotations (under PEP 649). In CPython we're calling this "annotation scopes", I'd suggest using the same here:
```suggestion
SCOPE_ANNOTATION: Final = 4  # Annotation scopes for type parameters and aliases (PEP 695)
```

Reply:
Renamed as suggested.